migrations/20231023212957_feeds.sql

```diff
@@ -0,0 +1,20 @@
+CREATE TABLE events(
+    id bigint NOT NULL PRIMARY KEY,
+    target_id bigint NOT NULL,
+    target_id_type text NOT NULL,
+    triggerer_id bigint NULL,
+    triggerer_id_type text NULL,
+    event_type text NOT NULL,
+    metadata jsonb NULL,
+    created timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+CREATE INDEX events_targets ON events (
+    target_id,
+    target_id_type,
+    event_type
+);
```

Comment:
what are event targets for? imo we should query events on the route based on the user's follows. events should just be registered in the database to be queried later.

Reply:
Maybe I need better names... how I modeled it is that an event _triggerer_ causes an event to happen to an event _target_.  So a triggerer would be someone like a user, organization, maybe an automatic process - whatever, it's the thing that caused the event to happen. Meanwhile a target is the thing the event happens to - for project-related events, it's the project. If we want to make it so someone can follow, idk, something like changes to organizations, the organization itself would be the target and the triggerer would be the user that initiated the change. The goal with this modeling was to reflect the fact that you might want to follow one (or both) sides of an event - you may be interested in what a user/organization is doing, or maybe you're just interested in one specific project.

I suppose we could drop the concept of a triggerer and insert two events for project creation - one event tied to the id of the user/org that did it and another tied to the id of the project or whatever, but that seems worse to me.

As for this specific line, it's just creating an index that matches how we query the table in an attempt to improve query performance.

Reply:
ah, that makes sense, I mistook this for a separate table